x/gateway/keeper/msg_server_stake_gateway_test.go

```diff
@@ -60,33 +61,61 @@ func TestMsgServer_StakeGateway_FailLoweringStake(t *testing.T) {
 	k, ctx := keepertest.GatewayKeeper(t)
 	srv := keeper.NewMsgServerImpl(k)
 
-	// Prepare the gateway
+	// Prepare the gateway.
 	addr := sample.AccAddress()
-	initialStake := sdk.NewCoin("upokt", math.NewInt(100))
-	stakeMsg := &types.MsgStakeGateway{
+	initialStake := cosmostypes.NewCoin("upokt", math.NewInt(100))
+	stakeMsg := &gatewaytypes.MsgStakeGateway{
 		Address: addr,
 		Stake:   &initialStake,
 	}
 
-	// Stake the gateway & verify that the gateway exists
+	// Stake the gateway & verify that the gateway exists.
 	_, err := srv.StakeGateway(ctx, stakeMsg)
 	require.NoError(t, err)
 	_, isGatewayFound := k.GetGateway(ctx, addr)
 	require.True(t, isGatewayFound)
 
-	// Prepare an updated gateway with a lower stake
-	updatedStake := sdk.NewCoin("upokt", math.NewInt(50))
-	updateMsg := &types.MsgStakeGateway{
+	// Prepare an updated gateway with a lower stake.
+	updatedStake := cosmostypes.NewCoin("upokt", math.NewInt(50))
+	updateMsg := &gatewaytypes.MsgStakeGateway{
 		Address: addr,
 		Stake:   &updatedStake,
 	}
 
-	// Verify that it fails
+	// Verify that it fails.
 	_, err = srv.StakeGateway(ctx, updateMsg)
 	require.Error(t, err)
 
-	// Verify that the gateway stake is unchanged
+	// Verify that the gateway stake is unchanged.
 	gatewayFound, isGatewayFound := k.GetGateway(ctx, addr)
 	require.True(t, isGatewayFound)
 	require.Equal(t, initialStake.Amount, gatewayFound.Stake.Amount)
 }
+
+func TestMsgServer_StakeGateway_FailBelowMinStake(t *testing.T) {
+	k, ctx := keepertest.GatewayKeeper(t)
+	srv := keeper.NewMsgServerImpl(k)
+
+	addr := sample.AccAddress()
+	gatewayStake := cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 100)
+	minStake := gatewayStake.AddAmount(math.NewInt(1))
+	expectedErr := gatewaytypes.ErrGatewayInvalidStake.Wrapf("gateway %q must stake at least %s", addr, minStake)
+
+	// Set the minimum stake to be greater than the gateway stake.
+	err := k.SetParams(ctx, gatewaytypes.Params{
```

Comment:
I'd prefer having this part as:
```
params := k.GetParams()
params.Minstake = minStake
k.SetParams(params)
```
I've been hit many times with overriding previous param values with their zero values.

It won't be the case here since gateway has only one param but I believe it's a good practice.